src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -701,21 +701,67 @@ impl ToBatch for LogicalAgg {
 
 impl ToStream for LogicalAgg {
     fn to_stream(&self) -> Result<PlanRef> {
+        let input = self.input();
+        // simple-agg
         if self.group_keys().is_empty() {
-            Ok(StreamSimpleAgg::new(
-                self.clone_with_input(
-                    self.input()
-                        .to_stream_with_dist_required(&RequiredDist::single())?,
-                ),
-            )
-            .into())
+            let mut has_min_or_max_agg_calls = false;
+            for c in &self.agg_calls {
+                match c.agg_kind {
+                    AggKind::Min | AggKind::Max => {
+                        has_min_or_max_agg_calls = true;
+                        break;
+                    }
+                    _ => continue,
+                }
+            }
+
+            let input_stream = input.to_stream()?;
+            let is_append_only = input_stream.append_only();
+            let input_distribution = input_stream.distribution();
+
+            // simple 2-phase-agg
+            if input_distribution.satisfies(&RequiredDist::AnyShard)
+                // TODO: non-append-only min/max is not supported yet,
+                // since min/max is a table rather than single value for local agg.
+                // see: https://github.com/singularity-data/risingwave/issues/2997#issuecomment-1156139576
+                && (is_append_only || !has_min_or_max_agg_calls)
```

Comment:
I'd prefer to have a standalone `allow_two_phase_agg` that takes all agg calls and `append_only` as parameter. In the future we will have more and more agg calls like `string_agg`, `count_distinct`, etc., and I'm not sure whether all of them can do two-phase. So having a single function for this can make it clearer for developers to know what is supported for now.